### PR TITLE
Various

### PR DESCRIPTION
**Pull request by zigasinko**

### CODE REVIEW

File: src/api/base.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
```
Comment (janybravo) [inline src/api/base.ts]: Should not be needed

File: src/api/csrfApi.ts [hunk 2]
```diff
@@ -4,6 +4,6 @@ import { get } from './fetch';
 const baseUrl = `${apiUrl}/base/csrf`;
 
 // eslint-disable-next-line import/prefer-default-export
-export const getCSRFCookie = (): Promise<unknown> => {
-    return get(baseUrl);
+export const getCSRFCookie = (): void => {
```
Comment (janybravo) [inline src/api/csrfApi.ts]: Please do return promise.

File: .eslintrc.js [hunk 3]
```diff
@@ -58,6 +59,13 @@ module.exports = {
         ],
         'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
         'prettier/prettier': 'error',
+        'react/require-default-props': 'off',
+        'no-void': ['error', { allowAsStatement: true }],
+        '@typescript-eslint/no-unsafe-member-access': 'off',
+        '@typescript-eslint/no-unsafe-assignment': 'off',
+        '@typescript-eslint/no-unsafe-call': 'off',
+        '@typescript-eslint/no-unsafe-return': 'off',
+        '@typescript-eslint/restrict-template-expressions': ['error', { allowAny: true }],
```
Comment (janybravo) [inline .eslintrc.js]: 1. add `'import/prefer-default-export': 'off'`
2. add `'@typescript-eslint/ban-ts-comment': 'off'`

File: src/api/fetch.ts [hunk 4]
```diff
@@ -3,6 +3,7 @@ import { getCookie } from '../utils/documentHelpers';
 import { sessionId } from './base';
 
 type QueryParams = { [key: string]: string | number };
+// eslint-disable-next-line @typescript-eslint/ban-types
```
Comment (janybravo) [inline src/api/fetch.ts]: Its better to have some type like: `Record<string, unknown>`

File: src/components/genexpress/common/reportBuilder/reportBuilder.ts [hunk 5]
```diff
@@ -79,5 +79,5 @@ export const exportToZip = async (prefix: string): Promise<void> => {
         logError('All exported files must have unique paths.');
     }
 
-    saveJsZip(createJsZip(uniqueFiles), prefix);
+    void saveJsZip(createJsZip(uniqueFiles), prefix);
```
Comment (janybravo) [inline src/components/genexpress/common/reportBuilder/reportBuilder.ts:82]: Its redundant

File: src/components/genexpress/genexpressAppBar/genexpressAppBar.tsx [hunk 6]
```diff
@@ -95,7 +95,7 @@ const GenexpressAppBar = ({
      * Execute export once user clicks on Export button in export prefix modal.
      */
     const handleExportPrefix = (prefix: string): void => {
-        reportBuilder.exportToZip(prefix);
+        void reportBuilder.exportToZip(prefix);
```
Comment (janybravo) [inline src/components/genexpress/genexpressAppBar/genexpressAppBar.tsx:98]: redundant

File: src/components/genexpress/modules/differentialExpressions/differentialExpressionsVolcanoPlot.tsx [hunk 7]
```diff
@@ -423,7 +423,7 @@ const DifferentialExpressionsVolcanoPlot = forwardRef(
             const highestValueInData = _.max(
                 _.map(
                     data,
-                    // eslint-disable-next-line @typescript-eslint/camelcase, no-restricted-globals
+                    // eslint-disable-next-line no-restricted-globals
```
Comment (janybravo) [inline src/components/genexpress/modules/differentialExpressions/differentialExpressionsVolcanoPlot.tsx]: use `Number.isFinite`

File: src/components/genexpress/modules/genesExpressions/genesExpressionsLineChart.tsx [hunk 8]
```diff
@@ -130,6 +132,12 @@ const getVegaSpecification = (
                   title: 'Genes',
                   stroke: 'colorscale',
                   orient: 'right',
+                  // 44 = offset of first gene in legend
+                  // 15.5 = gene name element height
+                  columns:
+                      chartHeight != null
```
Comment (janybravo) [inline src/components/genexpress/modules/genesExpressions/genesExpressionsLineChart.tsx]: invert expression

File: src/components/genexpress/modules/timeSeriesAndGeneSelector/geneSelector/geneSelector/geneSelector.tsx [hunk 9]
```diff
@@ -74,6 +74,9 @@ const GeneSelector = ({
 
     const isDisabled = _.isEmpty(basketInfo);
 
+    // Latest version of eslint (7.19.0) has a problem inferring dependencies from non-wrapped
```
Comment (janybravo) [inline src/components/genexpress/modules/timeSeriesAndGeneSelector/geneSelector/geneSelector/geneSelector.tsx]: its should start with  `TODO:`

**Review by janybravo — CHANGES_REQUESTED** [review]

File: src/api/base.ts (hunk 1 above)
Comment (zigasinko) [reply src/api/base.ts]: Node process variables are possibly undefined. I've enabled the rule back and used a different syntax
``export const apiUrl = `${process.env.REACT_APP_API_URL as string}`;``.

File: .eslintrc.js (hunk 3 above)
Comment (zigasinko) [reply .eslintrc.js]: Added both rules and removed all of their usages elseware.

File: src/api/fetch.ts (hunk 4 above)
Comment (zigasinko) [reply src/api/fetch.ts]: Yes, thanks.

File: src/components/genexpress/common/reportBuilder/reportBuilder.ts (hunk 5 above)
Comment (zigasinko) [reply src/components/genexpress/common/reportBuilder/reportBuilder.ts:82]: Left it in in accordance with eslint rule:  [https://github.com/typescript-eslint/typescript-eslint/blob/v4.14.2/packages/eslint-plugin/docs/rules/no-floating-promises.md]()

File: src/components/genexpress/genexpressAppBar/genexpressAppBar.tsx (hunk 6 above)
Comment (zigasinko) [reply src/components/genexpress/genexpressAppBar/genexpressAppBar.tsx:98]: Left it in in accordance with eslint rule:  [https://github.com/typescript-eslint/typescript-eslint/blob/v4.14.2/packages/eslint-plugin/docs/rules/no-floating-promises.md]()

File: src/components/genexpress/modules/differentialExpressions/differentialExpressionsVolcanoPlot.tsx (hunk 7 above)
Comment (zigasinko) [reply src/components/genexpress/modules/differentialExpressions/differentialExpressionsVolcanoPlot.tsx]: Nice one!

File: src/components/genexpress/modules/timeSeriesAndGeneSelector/geneSelector/geneSelector/geneSelector.tsx (hunk 9 above)
Comment (zigasinko) [reply src/components/genexpress/modules/timeSeriesAndGeneSelector/geneSelector/geneSelector/geneSelector.tsx]: Marked the comment with `TODO`.

**Review by janybravo — APPROVED** [review]

**Pull request merged**